Styleguide lists

mparke opened the pull request:
closes #892 
closes #903 

![screen shot 2017-03-16 at 10 26 54 am](https://cloud.githubusercontent.com/assets/709951/24000950/20b3ff1a-0a33-11e7-9dae-aa0d8ab55570.png)

mparke commented on src/linodes/layouts/IndexPage.js:
```diff
@@ -108,50 +118,71 @@ export class IndexPage extends Component {
+          return (
+            <div key={index}>
+              <div className="GroupLabel">{group.name}</div>
+              <Table
```
Example of the table interface

eatonphil replied:
-1 for breaking out groups into different tables if this does not also make sure that tables in the same group all share the same column widths when columns widths can be different.

mparke commented on src/linodes/layouts/IndexPage.js:
```diff
@@ -108,50 +118,71 @@ export class IndexPage extends Component {
+          name: _group,
+          // TODO: don't redefine these for each
+          columns: [
+            { cellComponent: CheckboxCell },
```
example of custom cell rendering

eatonphil commented:
I was hoping to see changes to the configs table too. Is that coming?

mparke commented:
TODO remaining:

- [ ] Node balancers list
- [ ] Authorized application scopes list
- [ ] Linode configs list
- [ ] Linode backups list
- [ ] Tests
- [ ] Linting

eatonphil commented on src/components/tables/Table.js:
```diff
@@ -0,0 +1,72 @@
+
+  let tableContent;
+  if (!data.length) {
+    tableContent = (<p>No records found.</p>);
```
Not a big deal, but this is probably not the default message we'd want.

na3d replied:
Default is dependent on the usage, no?

eatonphil commented on src/components/tables/cells/AnchorCell.js:
```diff
@@ -0,0 +1,38 @@
+        className="Table-rowLabel"
+        onClick={onClick}
+        title={record[titleKey]}
+      >
```
I don't think we want to be using anchor tags directly anywhere in the source. Use Link instead?

na3d replied:
I think I have seen anchors used elsewhere, we may need to go through and do some clean up.

eatonphil commented on src/components/tables/cells/IPRdnsCell.js:
```diff
@@ -0,0 +1,29 @@
+  const ip = ipKey ? record[ipKey] : record;
+  let str = `${ip.address}`;
+  if (ip.rdns) {
+    str += `(${ip.rdns})`;
```
Missing a space here before the (?

eatonphil commented on src/linodes/linode/networking/components/IPList.js:
```diff
@@ -1,32 +1,26 @@
+          { dataKey: 'address', label: 'IP Address' },
+        ]}
+        data={transferableIps}
+        selected={checked}
```
This used to be a secondary table?

eatonphil commented on src/linodes/linode/networking/components/IPSharing.js:
```diff
@@ -104,10 +104,20 @@ export default class IPSharing extends Component {
-              labels={['IP Address', 'Linode', '']}
-              keys={['address', 'linode', '']}
-              rows={rows}
+            <Table
```
Used to be a secondary table?

mparke replied:
fixed

eatonphil commented on src/linodes/linode/networking/layouts/ReverseDNSPage.js:
```diff
@@ -76,10 +79,9 @@ export class ReverseDNSPage extends Component {
-          keys={keys}
-          rows={records}
+        <Table
+          columns={columns}
```
Used to be a secondary table?

eatonphil commented on src/nodebalancers/nodebalancer/layouts/IndexPage.js:
```diff
@@ -49,41 +48,33 @@ export class IndexPage extends Component {
-        labels={labels}
-        keys={keys}
-        rows={newConfigs}
+      <Table
```
Used to be a secondary table?

mparke replied:
fixed

eatonphil commented:
Overall looks good. The OAuth scopes are rendered a little funny now though:

![screen shot 2017-03-17 at 11 52 00 am](https://cloud.githubusercontent.com/assets/3925912/24051364/390bba50-0b08-11e7-9838-c5534b5abd39.png)

* [ ] .OAuthScopes used to have a padding top 15px
* [ ] The scope title used to be capitalized

eatonphil commented:
Some padding issues with the buttons here and the table headers should not be bolded:

![screen shot 2017-03-17 at 11 54 29 am](https://cloud.githubusercontent.com/assets/3925912/24051486/9108adee-0b08-11e7-9a7c-0a9cca3dc6ae.png)

eatonphil commented:
But again, -1 to reverting to multiple tables in a group:

![screen shot 2017-03-17 at 11 57 36 am](https://cloud.githubusercontent.com/assets/3925912/24051612/06170310-0b09-11e7-80d4-56ee71f6eda6.png)

mparke commented:
@eatonphil re: groups and tables, they need to be separate at this stage, forcing group definitions into a table is additional complexity. It is 1 table / group.

mparke commented:
will see if I can fix some of the css problems

mparke commented:
Fixed some css problems, rebased, fixed tests and linting

![screen shot 2017-03-17 at 2 14 25 pm](https://cloud.githubusercontent.com/assets/709951/24057000/284039c6-0b1c-11e7-83b8-2eb0a9299a76.png)
![screen shot 2017-03-17 at 2 14 43 pm](https://cloud.githubusercontent.com/assets/709951/24056999/284022d8-0b1c-11e7-8ca1-11c641d2fd0c.png)
![screen shot 2017-03-17 at 2 14 54 pm](https://cloud.githubusercontent.com/assets/709951/24057001/2840c418-0b1c-11e7-9a1e-e4158c4bab32.png)

mparke commented:
Follow ups from TODOs:

https://github.com/linode/manager/issues/1410
https://github.com/linode/manager/issues/1411
https://github.com/linode/manager/issues/1412
https://github.com/linode/manager/issues/1413
https://github.com/linode/manager/issues/1414
https://github.com/linode/manager/issues/1415
https://github.com/linode/manager/issues/1416
https://github.com/linode/manager/issues/1417
https://github.com/linode/manager/issues/1418
https://github.com/linode/manager/issues/1419
https://github.com/linode/manager/issues/1420

@na3d

mparke commented:
and one more https://github.com/linode/manager/issues/1421
